vector/src/main/java/im/vector/fragments/VectorSettingsPreferencesFragment.kt

```diff
@@ -1148,8 +1148,29 @@ class VectorSettingsPreferencesFragment : PreferenceFragmentCompat(), SharedPref
         } else {
             removeDeactivateAccountPreference()
         }
+
+        mDefaultCompressionPreference.setOnPreferenceChangeListener { preference, newValue ->
+            changePreferenceSummary(mDefaultCompressionPreference, newValue.toString(), resources.getString(R.string.settings_compression_message_summary))
+            false
+        }
+        changePreferenceSummary(mDefaultCompressionPreference, mDefaultCompressionPreference.value, resources.getString(R.string.settings_compression_message_summary))
+
+        mDefaultMediaSourcePreference.setOnPreferenceChangeListener { preference, newValue ->
```

Comment:
Lines 1158 to 1162 are the same pattern as 1152 to 1156, I think the logic should be centralised into a method, to avoid typo mistakes.

Reply:
I am calling a function 2 times. one is for setting initial/existing value and another one is from the listener. The listener part will be called when the user changes the value. The same thing is actually applied for two different views with two different texts.

Reply:
Yes the two different view components and two different texts would be the parameters NOTE: I am unsure if the preference or newValue variables would also need to be parameters e.g.

```
   applyPreferenceSummary(mDefaultCompressionPreference, R.string.settings_compression_message_summary)
   applyPreferenceSummary(mDefaultMediaSourcePreference, R.string.settings_media_source_message_summary)


   private fun applyPreferenceSummary(listPreference: ListPreference, settingName: String){
        listPreference.setOnPreferenceChangeListener { preference, newValue ->
            changePreferenceSummary(listPreference, newValue.toString(), resources.getString(settingName))
            false
        }
        changePreferenceSummary(listPreference, listPreference.value, resources.getString(settingName))
    }

```